cms/djangoapps/contentstore/api/views/course_validation.py

```diff
@@ -110,6 +118,23 @@ def _assignments_validation(self, course):
         assignments, visible_assignments = self._get_assignments(course)
         assignments_with_dates = [a for a in visible_assignments if a.due]
 
+        graded_assignments = self._get_graded_assignments(assignments)
```

Comment:
Is it necessary to return this information for both graded and non-graded assignments?  Or do we only care about graded?  If the latter, you can add a URL parameter that specifies whether to only include graded assignments.  This would be similar to [what is done in course_quality](https://github.com/edx/edx-platform/blob/5a9f4ea3e61602d14215c53a3d4fa0885e1894ce/cms/djangoapps/contentstore/api/views/course_quality.py#L37).